src/main/java/io/frictionlessdata/datapackage/JSONBase.java

```diff
@@ -324,7 +332,7 @@ protected static String getZipFileContentAsString(Path inFilePath, String fileNa
         }
     }
 
-    public static JSONObject dereference(File fileObj, Path basePath, boolean isArchivePackage) throws IOException {
+    public static ObjectNode dereference(File fileObj, Path basePath, boolean isArchivePackage) throws IOException {
```

Comment:
Same here, this should return a POJO, not a JSON node (I'll stop here because the same comment applies to so many other places in the code).